src/app/domain/repositories/base.py

```diff
@@ -0,0 +1,25 @@
+import uuid
+from typing import Dict
+
+from django.db.models import Q
+
+
+class BaseRepository:
+    def find(self, id: uuid.UUID):
+        try:
+            return self.model.objects.get(id=id)
+        except self.model.DoesNotExist:
+            return False
+
+    def find_by_query(self, query):
+        return self.model.objects.filter(query)
+
+    def create(self, data: Dict):
+        return self.model.objects.create(**data)
+
+    def update(self, id: uuid.UUID, data: Dict, query=Q()):
+        self.model.objects.filter(id=id).filter(query).update(**data)
+        return self.find(id)
```

Comment:
Não recomendo fazer a atualização dessa forma.

Utilizar o `update` de um QuerySet faz com que não seja chamado os signals dos models.

E não há nenhum ganho de performance em utilizar dessa forma. Inclusive existe a consulta adicional para poder retornar a nova instância do model com os dados atualizados!

```suggestion
    def update(self, id: uuid.UUID, data: Dict, query=Q()):
        self.model.objects.filter(id=id).filter(query).update(**data)
        return self.find(id)
```